utils/keycodes.js

```diff
@@ -38,99 +44,49 @@ export function isMacOS( _window = window ) {
 	return _window.navigator.platform.indexOf( 'Mac' ) !== -1;
 }
 
-/**
- * Create a keyboard shortcut based on a string of modifiers + key(s).
- *
- * This function is not intended to be used directly by developers.
- * Instead, use primaryShortcut(), accessShortcut(), etc.
- *
- * @param {string} keys          Modifier and keyboard keys, seperated by "+".
- * @param {Object} _isMacOS      isMacOS function by default; used for DI testing.
- *
- * @return {string}          The keyboard shortcut.
- */
-export function keyboardShortcut( keys, _isMacOS = isMacOS ) {
-	const isMac = _isMacOS();
-
-	const alt = isMac ? '⌥option' : 'Alt';
-	const meta = isMac ? '⌃control' : '⊞';
-	const primary = isMac ? '⌘' : 'Ctrl';
-	const shift = isMac ? '⇧shift' : 'Shift';
-
-	const replacementKeyMap = {
-		[ ALT ]: alt,
-		[ META ]: meta,
-		[ PRIMARY ]: primary,
-		[ SHIFT ]: shift,
-	};
-
-	return keys
-		.replace( /\s/g, '' )
-		.split( '+' )
-		.map( ( key ) => {
-			return replacementKeyMap.hasOwnProperty( key ) ?
-				replacementKeyMap[ key ] : key;
-		} )
-		.join( '+' )
-		// Because we use just the clover symbol for MacOS's "command" key, remove
-		// the key join character ("+") between it and the final character if that
-		// final character is alphanumeric. ⌘S looks nicer than ⌘+S.
-		.replace( /⌘\+([a-zA-Z0-9])$/g, '⌘$1' );
-}
-
-/**
- * Create an access key shortcut based on a single character.
- *
- * Access key combo is:
- *  - Control+Alt on MacOS.
- *  - Shift+Alt on Windows/everywhere else.
- *
- * @param {string} character The character for the access combination.
- * @param {Object} _isMacOS  isMacOS function by default; used for DI testing.
- *
- * @return {string}          The keyboard shortcut.
- */
-export function accessShortcut( character, _isMacOS = isMacOS ) {
-	return keyboardShortcut( accessKeyCode( character.toUpperCase(), _isMacOS ), _isMacOS );
-}
-
-export function accessKeyCode( character, _isMacOS = isMacOS ) {
-	const keyCombo = _isMacOS() ? `${ META }+${ ALT }` : `${ SHIFT }+${ ALT }`;
-	return `${ keyCombo }+${ character }`;
-}
+const modifiers = {
+	primary: ( _isMac ) => _isMac() ? [ COMMAND ] : [ CTRL ],
```

Comment:
Why do these need to be run as functions? Will they ever change over the lifetime of the app?

**Edit:** And if the answer is for testing, it shouldn't really impact the implementation so much. Can be "good" for it to be made generic, though ideally we have some caching along the way in runtime use, so we're not running `_window.navigator.platform.indexOf( 'Mac' ) !== -1;` hundreds of times per page session.

Reply:
It's all for testing; previously this was run once and stored in a constant, but it was also entirely untested. It could probably be cached somehow as a perf-versus-readability tradeoff. 😄 

It shouldn't change in regular usage though, so yeah it's worth doing.